### PR TITLE
OP-438 New HTTP Client for SMS Gateway

### PR DESCRIPTION
**Pull request by AndreiDodu**
configuration (for core project)

Corrected some issues following suggestions of @dbmalkovsky and @mwithi

Merged informaticidevelope with my changes
In src/main/java/org/isf/menu/gui/MainMenu.java, Line: 325 -> please update TITLE with correct value

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@ logs
 *.iml
 *.bak
 *.pdf
+/pom.xml
```
Comment (mwithi) [inline .gitignore:12]: changes to the pom.xml should not be ignored

File: rsc/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/generalData.properties
```
Comment (mwithi) [inline rsc/.gitignore:1]: changes to the generalData.properties should not be ignored (now it is named settings.properties)

File: rsc/settings.properties [hunk 3]
```diff
@@ -26,7 +26,7 @@ LABEXTENDED=yes
 LABMULTIPLEINSERT=yes
 INTERNALPHARMACIES=yes
 MERGEFUNCTION=yes
-SMSENABLED=no
+SMSENABLED=yes
```
Comment (mwithi) [inline rsc/settings.properties:29]: don't know yet if to enable it by default

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 4]
```diff
@@ -67,23 +64,20 @@
...
 
 	private static final long serialVersionUID = 7620582079916035164L;
-	public static final String ADMIN_STR = "admin";
-	private boolean flag_Xmpp;
-	private boolean flag_Sms;
+	private boolean flag_Xmpp = false;
+	private boolean flag_Sms = false;
 
-	private static final Logger LOGGER = LoggerFactory.getLogger(MainMenu.class);
+	private final Logger logger = LoggerFactory.getLogger(MainMenu.class);
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:70]: correct version is with initial values and static LOGGER

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 5]
```diff
@@ -67,23 +64,20 @@
...
 
-	@Override
 	public void loginInserted(AWTEvent e) {
 		if (e.getSource() instanceof User) {
 			myUser = (User) e.getSource();
 			MDC.put("OHUser", myUser.getUserName());
 			MDC.put("OHUserGroup", myUser.getUserGroupName().getCode());
-			LOGGER.info("Logging: \"{}\" user has logged into the system.", myUser.getUserName());
+			logger.info("Logging: \"{}\" user has logged the system.", myUser.getUserName());
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:77]: correct form is the one on develop LOGGER and "into"

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 6]
```diff
@@ -92,15 +86,17 @@ public void commandInserted(AWTEvent e) {
 
 	public static boolean checkUserGrants(String code) {
 
-		for (UserMenuItem umi : myMenu) {
+		Iterator<UserMenuItem> it = myMenu.iterator();
+		while (it.hasNext()) {
+			UserMenuItem umi = it.next();
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:91]: correct form was with implicit iterator

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 7]
```diff
@@ -110,33 +106,33 @@ public int getMinButtonSize() {
...
-		Locale.setDefault(new Locale(GeneralData.LANGUAGE)); //for all fixed options YES_NO_CANCEL in dialogs
+		Locale.setDefault(new Locale(GeneralData.LANGUAGE)); // for all fixed options YES_NO_CANCEL in dialogs
 		singleUser = GeneralData.getGeneralData().getSINGLEUSER();
 		MessageBundle.getBundle();
 		try {
 			internalPharmacies = GeneralData.INTERNALPHARMACIES;
 			debug = GeneralData.DEBUG;
 			if (debug) {
-				LOGGER.info("Debug: OpenHospital in debug mode.");
+				logger.info("Debug: OpenHospital in debug mode.");
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:135]: correct form is with LOGGER

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 8]
```diff
@@ -152,16 +148,16 @@ public MainMenu() {
 		}
 
 		if (singleUser) {
-			LOGGER.info("Logging: Single User mode.");
-			myUser = new User(ADMIN_STR, new UserGroup(ADMIN_STR, ""), ADMIN_STR, "");
+			logger.info("Logging: Single User mode.");
+			myUser = new User("admin", new UserGroup("admin", ""), "admin", "");
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:152]: correct form is with LOGGER and constants

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 9]
```diff
@@ -152,16 +148,16 @@ public MainMenu() {
...
 			MDC.put("OHUser", myUser.getUserName());
 			MDC.put("OHUserGroup", myUser.getUserGroupName().getCode());
 		} else {
 			// get an user
-			LOGGER.info("Logging: Multi User mode.");
+			logger.info("Logging: Multi User mode.");
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:157]: correct form is with LOGGER

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 10]
```diff
@@ -152,16 +148,16 @@ public MainMenu() {
...
 			new Login(this);
 
-			if (null == myUser) {
+			if (myUser == null) {
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:160]: new standard on develop is to put values as first operand (to avoid NPE)

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 11]
```diff
@@ -180,26 +176,25 @@ public MainMenu() {
 				Server.getInstance().login(myUser.getUserName(), myUser.getPasswd());
 				try {
 					Thread.sleep(500);
-				} catch (InterruptedException interruptedException) {
-					LOGGER.error(interruptedException.getMessage(), interruptedException);
+				} catch (InterruptedException e) {
+					e.printStackTrace();
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:180]: correct form is with interruptedException and LOGGER.error

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 12]
```diff
@@ -180,26 +176,25 @@ public MainMenu() {
...
 					} else {
-						LOGGER.error("Passwords do not match, please drop the XMPP user and login to OH again with the same user.");
+						logger.error("Passwords not matching, please drop XMPP user and login OH again with the same user");
 					}
 				} else if (message.contains("XMPPError connecting")) {
-					LOGGER.error("No XMPP Server seems to be running: set XMPPMODULEENABLED = false");
+					logger.error("No XMPP Server seems to be running: set XMPPMODULEENABLED = false");
 				} else {
-					LOGGER.error("An error occurs: {}", e.getMessage());
+					logger.error("An error occurs: {}", e.getMessage());
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:197]: correct form was one on develop

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 13]
```diff
@@ -209,30 +204,36 @@ public MainMenu() {
 		// if in singleUser mode remove "users" and "communication" menu
 		if (singleUser) {
 			ArrayList<UserMenuItem> junkMenu = new ArrayList<>();
-			for (UserMenuItem umi : myMenu) {
-				if ("USERS".equalsIgnoreCase(umi.getCode()) || "USERS".equalsIgnoreCase(umi.getMySubmenu())) {
+			Iterator<UserMenuItem> it = myMenu.iterator();
+			while (it.hasNext()) {
+				UserMenuItem umi = it.next();
+				if (umi.getCode().equalsIgnoreCase("USERS") || umi.getMySubmenu().equalsIgnoreCase("USERS"))
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:210]: correct form was with implicit iterator

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 14]
```diff
@@ -209,30 +204,36 @@ public MainMenu() {
...
 					junkMenu.add(umi);
-				}
-				if ("communication".equalsIgnoreCase(umi.getCode())) {
+				if (umi.getCode().equalsIgnoreCase("communication")) {
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:212]: new standard on develop is with value as first operand (to avoid NPE)

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 15]
```diff
@@ -209,30 +204,36 @@ public MainMenu() {
...
 					if (flag_Xmpp) {
-						LOGGER.info("Single user mode: set XMPPMODULEENABLED = false");
+						logger.info("Single user mode: set XMPPMODULEENABLED = false");
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:214]: correct form is with LOGGER

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 16]
```diff
@@ -209,30 +204,36 @@ public MainMenu() {
...
 						flag_Xmpp = GeneralData.XMPPMODULEENABLED = false;
 					}
 					junkMenu.add(umi);
 				}
 			}
-			for (UserMenuItem umi : junkMenu) {
+			Iterator<UserMenuItem> altIt = junkMenu.iterator();
+			while (altIt.hasNext()) {
+				UserMenuItem umi = altIt.next();
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:222]: correct form is with implicit iterator

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 17]
```diff
@@ -209,30 +204,36 @@ public MainMenu() {
...
+				while (it.hasNext()) {
+					UserMenuItem umi = it.next();
+					if (umi.getCode().equalsIgnoreCase("communication"))
 						junkMenu.add(umi);
-					}
 				}
-				for (UserMenuItem umi : junkMenu) {
+				Iterator<UserMenuItem> altIt = junkMenu.iterator();
+				while (altIt.hasNext()) {
+					UserMenuItem umi = altIt.next();
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:236]: correct form is with implicit iterator

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 18]
```diff
@@ -241,28 +242,34 @@ public MainMenu() {
...
+			while (it.hasNext()) {
+				UserMenuItem umi = it.next();
+				if (umi.getCode().equalsIgnoreCase("MEDICALSWARD") || umi.getMySubmenu().equalsIgnoreCase("MEDICALSWARD"))
 					junkMenu.add(umi);
-				}
 			}
-			for (UserMenuItem umi : junkMenu) {
+			Iterator<UserMenuItem> altIt = junkMenu.iterator();
+			while (altIt.hasNext()) {
+				UserMenuItem umi = altIt.next();
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:253]: correct form is with implicit iterator and values as first operand (to avoid NPE)

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 19]
```diff
@@ -241,28 +242,34 @@ public MainMenu() {
...
+		while (it.hasNext()) {
+			UserMenuItem umi = it.next();
+			if (!umi.isActive())
 				junkMenu.add(umi);
-			}
 		}
-		for (UserMenuItem umi : junkMenu) {
+		Iterator<UserMenuItem> altIt = junkMenu.iterator();
+		while (altIt.hasNext()) {
+			UserMenuItem umi = altIt.next();
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:268]: correct form is with implicit iterators

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 20]
```diff
@@ -241,28 +242,34 @@ public MainMenu() {
...
 			myMenu.remove(umi);
 		}
 
-		setTitle(MessageBundle.formatMessage("angal.mainmenu.fmt.title", myUser.getUserName()));
+		setTitle(myUser.getUserName());
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:272]: correct form was the one on develop

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 21]
```diff
@@ -274,13 +281,15 @@ public MainMenu() {
...
+
 		int frameHeight = getSize().height;
+
 		setLocation(menuXPosition, screenHeight - frameHeight - menuYDisplacement);
 
 		myFrame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
 		myFrame.setAlwaysOnTop(GeneralData.MAINMENUALWAYSONTOP);
 		myFrame.addWindowListener(new WindowAdapter() {
-			@Override
+
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:292]: `@Override` is needed

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 22]
```diff
@@ -291,15 +300,12 @@ public void windowClosing(WindowEvent e) {
...
 
 	private void actionExit(int status) {
-		if (2 == status) {
-			LOGGER.info("Login failed.");
-		}
-		String newLine = System.lineSeparator();
-		LOGGER.info("{}{}====================={} Open Hospital closed {}====================={}", newLine, newLine, newLine, newLine, newLine);
+		if (status == 2)
+			logger.info("Login failed.");
+		logger.info("\n\n=====================\n Open Hospital closed \n=====================\n");
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:305]: correct form was the one on develop

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 23]
```diff
@@ -291,15 +300,12 @@ public void windowClosing(WindowEvent e) {
...
 		System.exit(status);
 	}
 
-	@Override
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:302]: `@Override` is needed

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 24]
```diff
@@ -312,17 +318,17 @@ private void launchApp(String itemMenuCode) {
 
 		for (UserMenuItem u : myMenu) {
 			if (u.getCode().equals(itemMenuCode)) {
-				if ("EXIT".equalsIgnoreCase(u.getCode())) {
+				if (u.getCode().equalsIgnoreCase("EXIT")) {
 					actionExit(0);
 				} else if (u.isASubMenu()) {
-					new SubMenu(this, u.getCode(), u.getButtonLabel(), myMenu);
+					// TODO update TITLE value
+					new SubMenu(this, u.getCode(), "TODO: TITLE", myMenu);
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:325]: the correct form is the one on develop

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 25]
```diff
@@ -312,17 +318,17 @@ private void launchApp(String itemMenuCode) {
...
 					break;
 				} else {
 					String app = u.getMyClass();
 					// an empty menu item
-					if ("none".equalsIgnoreCase(app)) {
+					if (app.equalsIgnoreCase("none"))
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:330]: new standard on develop is with values as first operand (to avoid NPE)

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 26]
```diff
@@ -334,8 +340,12 @@ private void launchApp(String itemMenuCode) {
...
 							}
 						}
-					} catch (InstantiationException | ClassNotFoundException | IllegalAccessException ie) {
-						LOGGER.error("Error instantiating menu item: '{}' with class '{}'.", u.getCode(), u.getMyClass());
+					} catch (InstantiationException ie) {
+						ie.printStackTrace();
+					} catch (IllegalAccessException iae) {
+						iae.printStackTrace();
+					} catch (ClassNotFoundException cnfe) {
+						cnfe.printStackTrace();
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:348]: correct form is with multiple catch and LOGGER

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 27]
```diff
@@ -348,61 +358,71 @@ private void launchApp(String itemMenuCode) {
...
-			for (UserMenuItem u : myMenu) {
-				if (u.getMySubmenu().equals("main")) {
+			for (UserMenuItem u : myMenu)
+				if (u.getMySubmenu().equals("main"))
 					numItems++;
-				}
-			}
+
+			// System.out.println(numItems);
+
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:372]: new standard on develop is to have explicit code blocks {} for loops and if statements and to remove commented System.out.println

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 28]
```diff
@@ -348,61 +358,71 @@ private void launchApp(String itemMenuCode) {
...
-					button[k] = new JButton(u.getButtonLabel());
-					button[k].setMnemonic(KeyEvent.VK_A + (u.getShortcut() - 'A'));
+					button[k - 1] = new JButton(u.getButtonLabel());
+
+					button[k - 1].setMnemonic(KeyEvent.VK_A + (int) (u.getShortcut() - 'A'));
 
-					button[k].addActionListener(parentFrame);
-					button[k].setActionCommand(u.getCode());
+					button[k - 1].addActionListener(parentFrame);
+					button[k - 1].setActionCommand(u.getCode());
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:384]: why this change on "k"?

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 29]
```diff
@@ -348,61 +358,71 @@ private void launchApp(String itemMenuCode) {
...
-				buttons.add(jButton);
-			}
-			SpringUtilities.makeCompactGrid(buttons, button.length, 1, 6, 6, 8, 10);
-			add(buttons, BorderLayout.EAST);
-		}
-	}
+			GridBagLayout layout = new GridBagLayout();
+			buttons.setLayout(layout);
+
+			final int insetsValue = 6;
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:398]: the new standard on develop uses SprintUtilities

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 30]
```diff
@@ -348,61 +358,71 @@ private void launchApp(String itemMenuCode) {
...
-			if (dimension.getWidth() + 10 <= titleStringWidth) {
-				dimension = new Dimension(titleStringWidth, (int) dimension.getHeight());
+		private void setButtonsSize(JButton[] button) {
+			int maxH = 0;
+			int maxMax = 0;
+			int maxMin = 0;
+			int maxPrf = 0;
+
+			for (int i = 0; i < button.length; i++) {
+				maxH = Math.max(maxH, button[i].getMaximumSize().height);
```
Comment (mwithi) [inline src/main/java/org/isf/menu/gui/MainMenu.java:414]: why this change on the button size?

File: src/main/java/org/isf/sms/gui/SmsEdit.java [hunk 31]
```diff
@@ -283,55 +286,56 @@ private JPanel getJButtonPanel() {
...
 					String number = jNumberTextField.getText().replaceAll(" ", "").trim(); //$NON-NLS-1$ //$NON-NLS-2$
 					String text = jTextArea.getText();
 					Date schedDate = jSchedDateChooser.getDate();
-					
+
 					Sms smsToSend = new Sms();
 					smsToSend.setSmsNumber(number);
 					smsToSend.setSmsDateSched(schedDate);
 					smsToSend.setSmsUser(UserBrowsingManager.getCurrentUser());
+					System.err.println("SMS_USER: " + UserBrowsingManager.getCurrentUser());
```
Comment (mwithi) [inline src/main/java/org/isf/sms/gui/SmsEdit.java:305]: we should use LOGGER here

File: src/main/java/org/isf/sms/gui/SmsEdit.java [hunk 32]
```diff
@@ -283,55 +286,56 @@ private JPanel getJButtonPanel() {
...
 							StringBuilder message = new StringBuilder();
-							message.append(e1.getMessages().get(0).getMessage())
-								.append("\n")
-								.append(MessageBundle.getMessage("angal.sms.doyouwanttosplitinmoremessages"))
-								.append(" (").append(textParts).append(")?");
-							
+							message.append(e1.getMessages().get(0).getMessage()).append("\n")
+											.append(MessageBundle.getMessage("angal.sms.doyouwanttosplitinmoremessages")).append(" (").append(textParts)
+											.append(")?");
+
```
Comment (mwithi) [inline src/main/java/org/isf/sms/gui/SmsEdit.java:323]: unnecessary new lines

File: src/main/java/org/isf/sms/gui/SmsEdit.java [hunk 33]
```diff
@@ -283,55 +286,56 @@ private JPanel getJButtonPanel() {
...
 									OHServiceExceptionUtil.showMessages(e2, SmsEdit.this);
 									return;
 								}
-								
-							} else return;
-							
+
+							} else
+								return;
+
```
Comment (mwithi) [inline src/main/java/org/isf/sms/gui/SmsEdit.java:338]: unnecessary new lines

**Review by mwithi — CHANGES_REQUESTED** [review]
Mostly unwanted changes on MainMenu.java that are rolling back latest develop improvements

File: rsc/sms.properties [hunk 34]
```diff
@@ -1,11 +1,38 @@
...
+gsm-gateway-service.csmp=AT+CSMP\=17,167,0,0\r
+gsm-gateway-service.cmgf=AT+CMGF\=1\r
+gsm-gateway-service.gmm=AT+GMM\r\n 
+gsm-gateway-service.cmgs=AT+CMGS\="
+
+
+##################################################################
+# Skebby configuration
+##################################################################
+# first we will try to use token, if null, then we will try to login with the provided username and password and retrieve token
```
Comment (mwithi) [inline rsc/sms.properties:25]: where is token here?

File: src/main/java/org/isf/menu/gui/MainMenu.java [hunk 35]
```diff
@@ -92,15 +86,17 @@ public void commandInserted(AWTEvent e) {
...
+			UserMenuItem umi = it.next();
 			if (umi.getCode().equalsIgnoreCase(code)) {
 				return true;
 			}
 		}
 		return false;
 	}
 
-	private int minButtonSize;
+	private int minButtonSize = 0;
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/menu/gui/MainMenu.java:99]: One doesn't have to init values to 0 or null as that is the default; doing the init again is redundant and unnecessary.

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 8 above)
Comment (dbmalkovsky) [reply src/main/java/org/isf/menu/gui/MainMenu.java:152]: And why undo the "ADMIN_STR"?; that saves allocation of string literals and makes changing the value easier if ever needed.

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 12 above)
Comment (dbmalkovsky) [reply src/main/java/org/isf/menu/gui/MainMenu.java:197]: yes; the messages are worded better in develop

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 29 above)
Comment (dbmalkovsky) [reply src/main/java/org/isf/menu/gui/MainMenu.java:398]: the version in develop is the new look-n-feel

File: rsc/.gitignore (hunk 2 above)
Comment (AndreiDodu) [reply rsc/.gitignore:1]: I corrected

File: rsc/settings.properties (hunk 3 above)
Comment (AndreiDodu) [reply rsc/settings.properties]: You are true. Done.

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 4 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: I restored the current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 5 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: I restored the current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 29 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: I think that is an old code. I haven't dealt yet with the GUI editing.

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 30 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: I think that is an old code. I haven't dealt yet with the GUI editing.

File: src/main/java/org/isf/sms/gui/SmsEdit.java (hunk 33 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/sms/gui/SmsEdit.java]: I restored the current informatici version

File: rsc/sms.properties (hunk 34 above)
Comment (AndreiDodu) [reply rsc/sms.properties]: I took a look to the documentation of skebby. I couldn't test Skebby (they didn't approved my registration, perhaps, for two times with different accounts), but I think that USER_KEY and ACCESS_TOKEN are costants, so that they will never change for a user (there are certainly exceptions). I don't know what could be the value of USER_KEY, but I think that it will never change. I added two new properties USER_KEY and ACCESS_TOKEN. If they exists both, then every time we will try to send sms with Skebby, we will load the authentication informations from a properties file and send sms with the values from properties. I don't know how to retrieve the USER_KEY without make an http request (perhaps we can find the USER_KEY in the user profile?).

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 35 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: It is not my code. Seems that I have only added and then removed this line:
//Object target = Context.getApplicationContext().getBean(Class.forName(app));
I will restore the previous version which is on infomatici.

File: src/main/java/org/isf/sms/gui/SmsEdit.java (hunk 32 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/sms/gui/SmsEdit.java]: I restored the current informatici version

File: src/main/java/org/isf/sms/gui/SmsEdit.java (hunk 31 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/sms/gui/SmsEdit.java]: I restored the current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 6 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: I restored the current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 8 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: I restored the current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 9 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: I restored the current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 35 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: I restored the current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 10 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: restored current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 11 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: restored current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 12 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: restored current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 25 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: restored current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 24 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: restored current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 23 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: restored current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 22 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: restored current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 21 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: restored current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 20 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: restored current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 19 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: restored current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 18 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: restored current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 26 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: restored current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 27 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: restored current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 17 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: restored current informatici version

File: src/main/java/org/isf/menu/gui/MainMenu.java (hunk 16 above)
Comment (AndreiDodu) [reply src/main/java/org/isf/menu/gui/MainMenu.java]: restored current informatici version

**Review by mwithi — APPROVED** [review]
